### PR TITLE
Carousel navigation focus

### PR DESCRIPTION
**Pull request by MewenLeHo**
Closes [#1084 ](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/1084)

### CODE REVIEW

File: scss/_carousel.scss [hunk 1]
```diff
@@ -127,6 +128,15 @@
     --#{$boosted-prefix}control-bg: #{$carousel-control-icon-active-bg}; // Boosted mod
     color: $carousel-control-color;
   }
+
+  &:focus:focus-visible,
+  &:focus {
```
Comment (louismaximepiton) [inline scss/_carousel.scss:133]: I think this selector could be upgraded with a `[data-focus-visible-added]` in order to not have the focus state on click

File: scss/_carousel.scss [hunk 2]
```diff
@@ -127,6 +128,15 @@
...
+    & > span {
+      outline: 3px solid $brand-orange;
```
Comment (louismaximepiton) [inline scss/_carousel.scss:135]: Think using rem could be better in the style page

File: scss/_carousel.scss [hunk 3]
```diff
@@ -208,12 +218,22 @@
     &:hover,
     &:focus {
```
Comment (louismaximepiton) [inline scss/_carousel.scss:219]: I guess that the weird behavior I can see is due to these selectors : 
- Focus the current page indicator
- There's a semi-moon that rolls over

**Review by louismaximepiton — CHANGES_REQUESTED** [review]
I think the js should be improved ? Maybe on Bs side

Focus on carousel doesn't seem to pause it
Focus on indicators is weird : the current page indicator is kinda scaled up

**Comment by MewenLeHo** [comment]
> I think the js should be improved ? Maybe on Bs side
> 
> Focus on carousel doesn't seem to pause it Focus on indicators is weird : the current page indicator is kinda scaled up

Should the carousel pauses on focus? Maybe. It may be useful for keyboard users.
What do you think about it @julien-deramond? It's a question for @Aniort and our accessibility experts.
However it's not the subject of this pull request.

**Comment by julien-deramond** [comment]
> Should the carousel pauses on focus? Maybe. It may be useful for keyboard users. What do you think about it @julien-deramond? It's a question for @Aniort and our accessibility experts. However it's not the subject of this pull request.

Not sure it should since you're focusing the arrow, not the carousel itself (the content). But I let @Aniort gives us his feedback.
Agree that's not something to take into account in this PR. As said by @louismaximepiton if we change this behavior let's propose it directly to Bootstrap.

File: scss/_carousel.scss [hunk 4]
```diff
@@ -104,6 +104,7 @@
   text-align: center;
   background: none;
   border: 0;
+  outline: 0;
```
Comment (julien-deramond) [inline scss/_carousel.scss:107]: Missing `// Boosted mod`

File: scss/_carousel.scss (hunk 3 above)
Comment (julien-deramond) [reply scss/_carousel.scss:221]: What's the status of this comment? Resolved?

File: scss/_variables.scss [hunk 5]
```diff
@@ -563,7 +563,8 @@ $container-padding-x: $grid-gutter-width !default;
 $border-width:                .125rem !default;
 $border-widths: (
   1: $border-width * .5,
-  2: $border-width
+  2: $border-width,
+  3: $border-width * 1.5
```
Comment (julien-deramond) [inline scss/_variables.scss:567]: By doing this you're adding a `.border-3` utility so it must be documented in https://deploy-preview-1242--boosted.netlify.app/docs/5.1/utilities/borders/#width by adding the following example: `<span class="border border-3"></span>`.

Note: an impact is observed where `.border-3` is used in the docs but is handle directly via https://github.com/twbs/bootstrap/pull/36476.

File: scss/_carousel.scss [hunk 6]
```diff
@@ -124,9 +125,20 @@
   }
 
   &:active {
-    --#{$boosted-prefix}control-bg: #{$carousel-control-icon-active-bg}; // Boosted mod
+    --#{$boosted-prefix}control-bg: #{$carousel-control-icon-active-bg};
     color: $carousel-control-color;
   }
+
+  &:focus:focus-visible,
```
Comment (julien-deramond) [inline scss/_carousel.scss:132]: Just out of curiosity, why do we need this `&:focus:focus-visible`? In other component just having `&:focus { &[data-focus-visible-added] {` seems sufficient.

File: scss/_carousel.scss [hunk 7]
```diff
@@ -208,12 +220,24 @@
...
+    }
 
+    &:hover {
+      transform: scale($carousel-indicator-hover-scale);
       &::before {
         transform: translate3d(-50%, -50%, 0) scale($carousel-indicator-active-scale);
       }
     }
+
+    &:focus {
```
Comment (julien-deramond) [inline scss/_carousel.scss:232]: :thinking: Buttons in `.carousel-indicators` are focusable?

**Review by julien-deramond — CHANGES_REQUESTED** [review]

File: scss/_variables.scss (hunk 5 above)
Comment (MewenLeHo) [reply scss/_variables.scss:567]: Done

**Pull request merged**